lib/src/folding/parsers/fallback.dart

```diff
@@ -99,30 +113,94 @@ class FallbackFoldableBlockParser extends TextFoldableBlockParser {
                 }
                 break;
               }
+              if (c.contains(char)) {
+                _isPossibleCommentSequence = true;
+              }
+            }
+
+            if (isMultilineCommentingEnabled) {
+              for (final c in multilineCommentSequences!) {
+                if (tail.endsWith(c.item1)) {
+                  if (!serviceCommentLines.contains(lineIndex)) {
+                    if (!foundImportTerminator) {
+                      startBlock(lineIndex, FoldableBlockType.multilineComment);
+                      _shouldEndMultilineComment = true;
+                    } else {
+                      // shouldn't start multiline comment block
+
+                      // class MyClass { /*
+                      // */
+                      // }
+                      _shouldEndMultilineComment = false;
+                    }
+                    _startedMultilineCommentWith = c.item1;
+                    _isInMultilineComment = true;
+                    break;
+                  }
+                }
+                if (c.item1.contains(char)) {
+                  _isPossibleCommentSequence = true;
+                }
+              }
+            }
+          }
+
+          if (isInMultilineComment) {
+            for (final c in multilineCommentSequences!) {
+              if (tail.endsWith(c.item2) &&
+                  _startedMultilineCommentWith == c.item1) {
+                if (!serviceCommentLines.contains(lineIndex)) {
+                  final blocksCountBefore = blocks.length;
+                  if (_shouldEndMultilineComment) {
+                    endBlock(lineIndex, FoldableBlockType.multilineComment);
+
+                    if (blocksCountBefore == blocks.length) {
+                      // in case firstLine == endLine
+                      setFoundSingleLineComment();
+                    }
+                  }
+
+                  _isInMultilineComment = false;
+                  _startedMultilineCommentWith = null;
+                  break;
+                }
+              }
             }
           }
 
-          if (!_foundSingleLineComment && !foundImport) {
+          if (!_foundSingleLineComment &&
+              !foundImport &&
+              !_isPossibleCommentSequence) {
             setFoundImportTerminator();
           }
```

Comment:
There was a problem that if the length of a comment sequence is more than 1 character, we won't mark it as a single line comment.
Because isFoundImportTerminator has more priority when submitting a line and after we found first character of the line that is not a full comment we would immediately  mark it as an import terminator